slack/resource_conversation.go

```diff
@@ -157,6 +175,58 @@ func resourceSlackConversationCreate(ctx context.Context, d *schema.ResourceData
 	return resourceSlackConversationRead(ctx, d, m)
 }
 
+func findExistingChannel(ctx context.Context, client *slack.Client, name string, isPrivate bool) (*slack.Channel, error) {
+	// find the existing channel. Sadly, there is no non-admin API to search by name,
+	// so we must search through ALL the channels
+	tflog.Info(ctx, "Looking for channel %s", map[string]interface{}{"channel": name})
+	paginationComplete := false
+	cursor := ""       // initial empty cursor to begin at start of list
+	var types []string // default value with empty list is "public_channel"
+	if isPrivate {
+		types = append(types, "private_channel")
+	}
+	for !paginationComplete {
+		channels, nextCursor, err := client.GetConversationsContext(ctx, &slack.GetConversationsParameters{
+			Cursor: cursor,
+			Limit:  200, // 100 is default, docs recommend no more than 200, but 1000 is the max
```

Comment:
this magic number is from the documentation of the slack api.  See, for instance, a similar usage in their golang library.
https://github.com/slack-go/slack/blob/master/users.go#L305